sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -154,6 +155,22 @@ class CacheManager extends Logging {
     }
   }
 
+  // Analyzes column statistics in the given cache data
+  private[sql] def analyzeColumnCacheQuery(
+      sparkSession: SparkSession,
+      cachedData: CachedData,
+      column: Seq[Attribute]): Unit = {
+    val relation = cachedData.cachedRepresentation
+    val (rowCount, newColStats) =
+      CommandUtils.computeColumnStats(sparkSession, relation, column)
+    val oldStats = relation.statsOfPlanToCache
+    val newStats = oldStats.copy(
+      rowCount = Some(rowCount),
+      attributeStats = AttributeMap((oldStats.attributeStats ++ newColStats).toSeq)
+    )
+    relation.statsOfPlanToCache = newStats
```

Comment:
This function is able to ignore that assumption because this only updates `relation.statsOfPlanToCache` instead of `cachedData` itself?

Reply:
Yea, I think so. Rather, it seems we need `@volatile` for `statsOfPlanToCache`?

Reply:
+1 for adding `@volatile` explicitly.